### PR TITLE
chore: update eslint/prettier configs

### PR DESCRIPTION
**Pull request by AykutSarac**
Updates the following:
* eslint
* prettier
* stylelint v14 -> v15

Adds
* eslint-plugin-lit
* eslint-plugin-wc

New eslint plugins helped me to notice & fix some errors at components. See [665a03c](https://github.com/Trendyol/baklava/pull/657/commits/665a03c501c6bfa6095bc0641348b4c4cef3f450).

### CODE REVIEW

**Review by ogunb — APPROVED** [review]

**Comment by AykutSarac** [comment]
> Should we disable [whitespace-sensitive formatting](https://prettier.io/blog/2018/11/07/1.15.0.html#whitespace-sensitive-formatting)?
> 
> Screenshot 2023-07-20 at 15 13 26
> Other than that, lgtm. Eline sağlık.

Each file seem to have its own parser according to Prettier, also this blog post is from 2018 so I have doubts it does have an effect on us. Unit tests also does pass.

**Comment by AykutSarac** [comment]
Added 2 new eslint plugins for web components & lit. Additionally they helped me to identify some errors at existing components and tests.

**Comment by AykutSarac** [comment]
> > Each file seem to have its own parser according to Prettier, also this blog post is from 2018 so I have doubts it does have an effect on us. Unit tests also does pass.
> 
> I meant as a styling choice. I don't really like current span formatting but, it also prevents unwanted spacing issue.
> 
> That being said, If the spacing doesn't matter, we can just add spaces before and after the text and it will format regularly. So, no big deal.

I see. Just to confirm, what is the option for this config?
I've tried `"htmlWhitespaceSensitivity": "ignore"` assuming it was the config you were referring to and it produced some unwanted results:
```tsx
const content = this.embedUrl
      ? html`<iframe src=${this.embedUrl}></iframe>`
      : html`<slot></slot>`;
```

```tsx
 const content = this.embedUrl
      ? html`
          <iframe src=${this.embedUrl}></iframe>
        `
      : html`
          <slot></slot>
        `;
```
...
```tsx
const inputSelectedOptions = html`<ul class="selected-options">
      ${this._selectedOptions.map(item => html`<li>${item.textContent}</li>`)}
    </ul>`;
```
```tsx
const inputSelectedOptions = html`
      <ul class="selected-options">
        ${this._selectedOptions.map(
          item =>
            html`
              <li>${item.textContent}</li>
            `
        )}
      </ul>
    `;
```

**Review by fatihhayri — APPROVED** [review]

**Review by erbilnas — APPROVED** [review]

**Review by buseselvi — APPROVED** [review]

File: .prettierrc.json [hunk 1]
```diff
@@ -1,11 +1,18 @@
 {
   "arrowParens": "avoid",
-  "bracketSpacing": true,
-  "quoteProps": "consistent",
-  "printWidth": 100,
+  "quoteProps": "preserve",
+  "printWidth": 95,
```
Comment (Enes5519) [inline .prettierrc.json:4]: Why did we give 95?

Comment (AykutSarac) [reply .prettierrc.json:4]: I've reviewed the files after the lint and overall print width of 100 was too long for our codebase.

**Review by muratcorlu — APPROVED** [review]

**Pull request merged**